python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -473,6 +473,8 @@ def __init__(
         dropna : bool, optional
             If True (default), do not include the "null" group.
         """
+        if cudf.get_option("mode.pandas_compatible"):
+            obj = obj.nans_to_nulls()
```

Comment:
What exactly is this doing? Are we basically saying that we don't want NaNs to propagate through subsequent elements in the scan and the way libcudf is designed NaNs propagate but nulls do not?

Reply:
Yup exactly. And generally for any groupby operation, not just scans, give NaN null semantics instead of non-null semantics in pandas compat mode.